### PR TITLE
Add test for Testing Farm timeout

### PR DESCRIPTION
**Pull request by phracek**
Add test for Testing Farm timeout.

Check if timeout is handled properly

### CODE REVIEW

**Review by jamacku — COMMENTED** [review]
@phracek You can lower the `timeout` to hit it faster.

File: plans/timeout_plan.fmf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+summary: Basic timeout test
+execute:
+    how: tmt
+    duration: 10s
```
Comment (zmiklank) [inline plans/timeout_plan.fmf:4]:
```suggestion
    duration: 60s
```
Shouldn't this be 60 seconds? Because the sleep takes 60 seconds.

Comment (phracek) [reply plans/timeout_plan.fmf:4]: No, because we need test, that TEsting Farm really timeouts.

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

File: .github/workflows/timeout-test.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        uses: actions/checkout@v4
+        with:
+          ref: "refs/pull/${{ github.event.issue.number }}/head"
+
+      - name: Run the tests
+        uses: ./
+        id: tf_results
+        with:
+          api_key: ${{ secrets.TF_PUBLIC_API_KEY }}
+          git_url: "https://github.com/sclorg/testing-farm-as-github-action"
```
Comment (jamacku) [inline .github/workflows/timeout-test.yml:26]:
```suggestion
```

**Comment by phracek** [comment]
@jamacku PTAL and Review it. Thanks

File: .github/workflows/timeout-test.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+          api_key: ${{ secrets.TF_PUBLIC_API_KEY }}
+          git_ref: "main"
+          tmt_plan_regex: "timeout_plan"
+          pull_request_status_name: "Timeout test"
+          update_pull_request_status: "true"
+          timeout: 10
+
+      - name: Check if log contains exceed limit
+        run: |
+          curl {{ steps.tf_results.outputs.test_log_url }} > results.log
```
Comment (jamacku) [inline .github/workflows/timeout-test.yml:34]:
```suggestion
          curl ${{ steps.tf_results.outputs.test_log_url }} > results.log
```

**Comment by phracek** [comment]
[test]

**Review by jamacku — APPROVED** [review]

**Pull request merged**